app-android/src/main/java/com/ustadmobile/port/android/view/PersonDetailFragment2.kt

```diff
@@ -79,31 +98,368 @@ class PersonDetailFragment2 : Fragment(){
+            modifier = Modifier
+                .width(70.dp))
+
+        Column {
+            Text(text = "Par")
+            Text(text = "Par")
+
+            if (clazz.clazzEnrolmentRole == ClazzEnrolment.ROLE_STUDENT){
+                Row (
+                    verticalAlignment = Alignment.CenterVertically
+                ){
+                    Image(painter = painterResource(id = R.drawable.ic_lens_black_24dp),
+                        contentDescription = null,
+                        colorFilter = ColorFilter.tint(color = Color.Gray),
+                        modifier = Modifier
+                            .size(12.dp))
+
+                    Spacer(modifier = Modifier.width(10.dp))
+
+                    Text(text = "Par")
+                }
+            }
+        }
+    }
+}
+
+@Composable
+private fun DetailFeild(
+    imageId: Int = 0,
+    valueText: String,
+    labelText: String,
+    onClick: () -> Unit = {}
+){
+    TextButton(
+        onClick = onClick
+    ){
+        Row{
+            if (imageId != 0){
+                Image(
+                    painter = painterResource(id = imageId),
+                    contentDescription = null,
+                    colorFilter = ColorFilter.tint(color = Color.Gray),
+                    modifier = Modifier
+                        .size(24.dp))
+            } else {
+                Box(modifier = Modifier.width(24.dp))
+            }
+
+            Spacer(modifier = Modifier.width(10.dp))
+
+            Column {
+                Text(valueText,
+                    style = Typography.h4,
+                    color = Color.Black)
+
+                Text(labelText,
+                    style = Typography.body2,
+                    color = Color.Gray)
+            }
+        }
+    }
+}
+
+@Composable
+private fun QuickActionButton(text: String, imageId: Int, onClick: () -> Unit){
+    TextButton(
+        modifier = Modifier.width(110.dp),
+        onClick = onClick
+    ){
+        Column(
+            horizontalAlignment = Alignment.CenterHorizontally
+        ){
+            Image(
+                painter = painterResource(id = imageId),
+                contentDescription = null,
+                colorFilter = ColorFilter.tint(color = colorResource(R.color.primaryColor)),
+                modifier = Modifier
+                    .size(24.dp))
+
+            Text(text.uppercase(),
+                style= Typography.h4,
+                color = colorResource(R.color.primaryColor))
+        }
+    }
+}
+
+@Composable
+private fun PersonDetailScreen(viewModel: PersonDetailViewModel) {
+    val uiState: PersonDetailUiState by viewModel.uiState.collectAsState(PersonDetailUiState())
+    PersonDetailScreen(uiState)
 }
 
 @Composable
 @Preview
 fun PersonDetailScreenPreview() {
-    PersonDetailScreen(
-        person = Person().apply {
-            firstNames = "Bob"
-        }
-    )
+    val uiState = PersonDetailUiState()
```

Comment:
Please add some data for the preview so it shows more - e.g. add name, phone number, email address, etc.